internal/events/events.go

```diff
@@ -153,7 +150,7 @@ func instantiateDriver(
 		return eventersql.BuildPostgreSQLDriver(ctx, cfg)
 	case constants.NATSDriver:
 		zerolog.Ctx(ctx).Info().Msg("Using NATS driver")
-		return nats.BuildNatsChannelDriver(cfg)
+		return nats.BuildNatsChannelDriver(&cfg.Nats)
```

Comment:
All the other constructors take the entire `cfg`, so that's the model I followed here.

Reply:
This has become a tedious problem to refactor. I have to change the signature of `BuildNatsChannelDriver` as `cloudEventsNatsAdapter` is not externally exposed. 

Either I change the signature of every driver function to take in not `serverconfig`, but only the relevant config (in order to follow a pattern), which again isn't pretty as it does not truly achieve separation from `serverconfig` as functions like `BuildPostgreSQLDriver` would still rely on components purely in `serverconfig` (like `cfg.SQLPubSub.AckDeadline`) which isn't there in `reminderconfig`. Either it would result in breaking the pattern or some form of duplication of the instantiation code. 

I see how this can slowly become just duplication of all instantiation code in order to keep minder and reminder separate, but I'm torn between keeping them separate (as minder is the main engine and reminder is a completely different service) v/s keeping them in the same code for ease of maintainability (and for de duplication - which again strikes the thought that this duplication would _not_ have been there if reminder was in another language). 

What do you think about it?

Reply:
I think it's fine to re-use the _implementation_ of the construction code in reminder, even as it runs as a separate service.  (For example, most Kubernetes controllers use the same informers library and construction client to create their connection to the K8s apiserver.)

So, my suggestion would be to embed a `serverconfig.EventConfig` into the `ReminderConfig`, rather than attempting to mirror only selected options into the reminder config.  This will mean that it's possible to have something meaningless like the Go channel in reminder; you can avoid that by adding a check in the top-level `ReminderConfig.Validate` that the event configuration isn't using the go module.

Reply:
(This pattern will also give you access to the flag-driven eventer, and future improvements in flagging that I'm planning -- using a central flag server, rather than repeating the flag file for each binary.)